### DISCUSSION TITLE
Cluster/HA support?

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [PurpleI2P/i2pd#2100] [Q&A]
I am not good with titles, sorry...

Basically, I was wondering if I should deploy i2pd as a `kind: DemonSet` or `kind: Deployment` in my k3s cluster and then wondered if it actually _does_ support clustering in the first place?

Thanks!

**Comment by yhaenggi** [comment]
a DaemonSet is something which is required to run on any node, so thats the wrong one already.
The only thing you can do with multiple replicas, is multihoming. It's kinda load-balancing of incoming traffic for your server tunnels.
There's no clustering/quorum/etc feature in i2pd itself.

**Reply by IngwiePhoenix** [reply]
> a DaemonSet is something which is required to run on any node, so thats the wrong one already.

True; but what if they all shared the same storage (i.e. NFS share)? I get your point though.

Is there some documentation for multihoming? This sounds very interesting!

**Reply by yhaenggi** [reply]
a NFS share is provided via a PV directly, or a PVC, so it just gets mounted into the pod.
But they cannot share the same storage! Each replica will have a different view of the netDB and node profiles. Each router also need unique keys (not server tunnel keys).
For multihoming, you just need to provide the same key.dat from your keys = entry.
Preferably, if you want to do it clean; have the whole i2pd config in a confimap and the keys in a secret, then mount them into the container as volumes.

**Comment by x64x2** [comment]
For k3s deployment, use `kind: Deployment` instead of `kind: DaemonSet`. i2pd supports clustering btw

**Comment by Vort** [comment]
I don't know what are you trying to achieve, but in any case please ensure that each i2pd router have unique keys (`router.keys`, `ntcp2.keys`, `ssu2.keys`).
